include/bout/field_accessor.hxx

```diff
@@ -39,9 +39,9 @@ struct BoutRealArray {
 
   /// Cast operators, so can be assigned to a raw pointer
   /// Note: Not explicit, so can be cast implicitly
-  operator BoutReal*() { return data; }
+  BOUT_HOST_DEVICE operator BoutReal*() { return data; }
```

Comment:
This is obviously wrong, but from reading the linked page it's apparent that this is because `data` is public -- making it `private` silences the warning. @bendudson Any reason we can't make `data` private?

Reply:
@ZedThree I think this was an issue with Raja or CUDA compiler not liking private variables. It may be resolved now.